module-5.md

```diff
@@ -12,36 +12,102 @@ Planned:      1.25 hour
+* Review your cell phone usage
+  * Identify your apps’ data usage
+  * Record top 3 apps that took up the most data
+  * Was this data cellular, or WiFi?
+* Public WiFi Speed Test
+  * Use a speed test tool to test the speed of the public WiFi, record the result.
+  * Is it faster or slower than the one you have at home? 
+  * Run the speed test again, record the result
+  
+* Discuss
+  * What unit is the data usage of your apps measured in?
+  * What does MB mean?
+  * What was your top app and how much MB did it use?
+  * Was that surprising?
+  * Would you be more/less surprised if it was cellular data? What if it was WiFi data? 
+    * Are different data sources worth more?
+    * Why? What is your cell data limit?
+  
+  * What was the unit that the WiFi speed was measured in?
+  * What does Mbps mean?
+  * How does this relate to the MB from apps?
+  * Why was the speed different between tests?
+  
+* Show Grafana functioning on a node, and describe what it is
+  * Access via web interface
+  * Show how to select nodes
+  
+* Break into groups
+  * Connect to the node
+  * Access the Grafana interface and look around
+  * Stream a music file that is located on a node's local webserver and observe the metrics
+  * Stream a video file that is located on a node's local webserver and observe the metrics
+  * Observe what metrics change if you do the same from a remote node (over the mesh)
+  * Can you measure both nodes at the same time?
+
+* Discuss results
+  * Did difficulty increase measuring two nodes at the same time?
+  * What would happen if you had a citywide mesh?
+  * Are some metrics more important than others?
+  * How do you represent a mesh network?
+    * Physical network topology (i.e. Toronto Mesh map)
+    * Virtual peering topology (i.e. fc00.org)
 
 ## 3: Discussion and reflection activity
 
-Q & A about sharing broken internet experiences.
+Q & A about sharing broken internet experiences, understanding failure points and why we measure
+  * How would you be affected if the internet went down at home?
+    - Do you have alternate sources to connect to the internet (i.e. cellphone/LTE)?
+    - Do you wait for a user to complain that the internet is down or can you be instantly notified by monitoring your network?
+  * Have you ever had internet issues that were not complete outages? 
+    - Packet losses, poor signal quality, how much load time can you tolerate
+    - What are the most popular services on your network? How many users are accessing it and what is the available bandwidth?
+    - Do you need to setup additional nodes to offload bandwidth if a current node is over taxed with traffic?
+    - Who are the heaviest users? Do you need to introduce bandwidth metering? How do you prioritize bandwidth and users?
+    - Does network traffic change throughout the day?
+  * Port Traffic - clients can communicate over ports and port monitoring may be required if there is excessive traffic
```

Comment:
The concept of ports may not have been discussed sufficiently up until now. This will need an explanation.

Reply:
I agrea, this may be to technical for this workshop.  I think give them something to think about and how we actually do it is another matter.

Althought "PORTS" can also be defined as interfaces and we do see those in Grafana so we could tie it in there.

Reply:
I think looking at interfaces on Grafana is a good exercise. It's probably unnecessary to bring in ports as a new concept.